src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.dev.cinema.service;
+
+import com.dev.cinema.exceptions.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        if (userFromDb.isEmpty()) {
+            throw new AuthenticationException("Incorrect email " + email);
+        }
+        User user = userFromDb.get();
+        if (isValid(user.getPassword(), password, user.getSalt())) {
+            return user;
+        }
+        throw new AuthenticationException("Incorrect password");
```

Comment:
Let's unite these two **_if_** blocks and throw the exception once

Reply:
fixed